obstore/python/obstore/fsspec.py

```diff
@@ -59,27 +72,48 @@ class AsyncFsspecStore(fsspec.asyn.AsyncFileSystem):
     fsspec-style object.
     """
 
-    cachable = False
+    # https://github.com/fsspec/filesystem_spec/blob/56054c0a30ceedab4c0e6a0f7e429666773baf6d/docs/source/features.rst#instance-caching
+    cachable = True
 
-    def __init__(
+    def __init__(  # noqa: PLR0913
```

Comment:
It seems you want a `protocol` parameter here, which sets the value of `protocol` onto `self`?